### PR TITLE
Many=true item validation support for primitive properties

### PR DESCRIPTION
**Pull request by ottonomy**
Relates to #63. Adds support for many=True flag (defaults to false) in validate_primitive_property task.

Before #63 is fully resolved, we've got to support the same thing for validate_id_property.

### CODE REVIEW

**Comment by ottonomy** [comment]
Have not yet checked if this is compatible with #60. They may step on each other's toes a bit. #60 should be merged first then this (probably manually after +1).

**Comment by ottonomy** [comment]
Fixed minor incompatibilities with code from PR #60 and added feature for many-item ID validation. This is ready for review and merge.

**Comment by mgylling** [comment]
+1

File: badgecheck/tasks/utils.py [hunk 1]
```diff
@@ -10,3 +10,14 @@ def task_result(success=True, message='', actions=None):
         actions = []
 
     return (success, message, actions,)
+
+
+def is_empty_list(value):
+    return isinstance(value, (tuple, list,)) and len(value) == 0
```
Comment (ViktorHaag) [inline badgecheck/tasks/utils.py]: Is there value in doing something like

```
import collections
...
return isinstance(value, collections.Sized) and len(value) == 0
```

Or one of the other ABCs in collections: `(collections.Sequence, collections.Set)`, for example?

File: badgecheck/tasks/utils.py [hunk 2]
```diff
@@ -10,3 +10,14 @@ def task_result(success=True, message='', actions=None):
...
+
+
+def abbreviate_value(value):
+    if len(str(value)) < 48:
+        return str(value)
+    else:
+        return str(value)[:48] + '...'
```
Comment (ViktorHaag) [inline badgecheck/tasks/utils.py]: When you abbreviate, does the abbreviation itself have to follow the abbrev rules? In which case you want `str(value)[:45] + '...'` maybe? (If you don't do this, do you risk an endlessly abbreviating thing that could just swallow up its contents in dot characters if called repeatedly?)

File: badgecheck/tasks/validation.py [hunk 3]
```diff
@@ -152,30 +153,43 @@ def validate_primitive_property(state, task_meta):
     prop_name = task_meta.get('prop_name')
     prop_type = task_meta.get('prop_type')
     prop_value = node.get(prop_name)
-    required = bool(task_meta.get('prop_required'))
+    required = bool(task_meta.get('required'))
 
-    if not prop_value and required:
+    if prop_value is None and required:
```
Comment (ViktorHaag) [inline badgecheck/tasks/validation.py]: This allows for non-truthy prop values like `[]` and  `{}` which would be valid JSON values but still falsey. I take it that's what we want?

**Comment by ViktorHaag** [comment]
Looks good -- I just had a few questions. Sorry for the delay in review.

**Pull request merged**